### PR TITLE
Validate requirements format & dependency name

### PR DESCRIPTION
**Pull request by segiddins**
So we dont accept a spec where loading from compact index info files will always fail

Verified only a single recently uploaded gem would fail validation (for the requirements), will manually clear that one attribute since it has been yanked

Many dependencies have invalid unresolved names (11217 distinct invalid ones), hence only validating that now on create

### CODE REVIEW

**Comment by segiddins** [comment]
I have no clue why the tests are failing without an obvious failure in CI, they're passing for me locally

File: lib/patterns.rb [hunk 1]
```diff
@@ -7,6 +7,9 @@ module Patterns
   ROUTE_PATTERN         = /#{ALLOWED_CHARACTERS}/
   LAZY_ROUTE_PATTERN    = /#{ALLOWED_CHARACTERS}?/
   NAME_PATTERN          = /\A#{ALLOWED_CHARACTERS}\z/
+  LETTERS_REGEXP        = /[a-zA-Z]+/
```
Comment (martinemde) [inline lib/patterns.rb:10]: Could drop the `+` at the end and maintain behavior and linear time.

File: lib/patterns.rb [hunk 2]
```diff
@@ -7,6 +7,9 @@ module Patterns
...
+  SPECIAL_CHAR_PREFIX_REGEXP = /\A[#{Regexp.escape(SPECIAL_CHARACTERS)}]+/
```
Comment (martinemde) [inline lib/patterns.rb:11]: Could drop the `+` at the end and maintain behavior and linear time.

**Review by martinemde — APPROVED** [review]
Nice, thanks for adding all this.

File: lib/patterns.rb (hunk 2 above)
Comment (segiddins) [reply lib/patterns.rb:11]: it should still be linear in input size with the `+`, no?

Comment (martinemde) [reply lib/patterns.rb:11]: Just simpler, the `+` does nothing here. That's all.

**Comment by segiddins** [comment]
Coverage regression is only in logger code, which is expected now that we disable logging on CI to drastically improve performance

**Pull request merged**